sdks/python/apache_beam/io/fileio.py

```diff
@@ -243,6 +249,57 @@ def process(
     yield ReadableFile(metadata, self._compression)
 
 
+@experimental()
+class MatchContinuously(beam.PTransform):
+  """Checks for new files for a given pattern every interval.
+
+  This ``PTransform`` returns a ``PCollection`` of matching files in the form
+  of ``FileMetadata`` objects.
```

Comment:
Java has an EmptyMatchTreatment. Should we add that here too? https://beam.apache.org/releases/javadoc/2.30.0/org/apache/beam/sdk/io/fs/EmptyMatchTreatment.html

Reply:
I thought about, but I am not sure this would have an actual use case. Not allowing EmptyMatches would produce an error, and since this is a streaming pipeline, that errors would be retried infinitely. 

https://beam.apache.org/releases/pydoc/2.30.0/_modules/apache_beam/io/fileio.html#EmptyMatchTreatment

`MatchAll` has the option to add it and wouldn't be hard, but I think it would translate in the PTransform being less resilient. What do you think?

Reply:
hmmm I think that's fair. Let's leave it as it is for now.